### DISCUSSION TITLE
Starting Valkey using systemd (Ubuntu) - compiled without libsystemd support?

### DISCUSSION THREAD

**Discussion by melroy89** [valkey-io/valkey#1103] [Q&A] [answered]
Hi,

I'm really eager to move away from Redis to Valkey. However after downloading the "Focal" (I'm actually running Ubuntu 24.04) Valkey v8.0.0 standalone binary from the official site: https://valkey.io/download/.

I setup the [systemd service file](https://github.com/valkey-io/valkey/blob/unstable/utils/systemd-valkey_server.service). I configured everything how I want Valkey to run. Everything seems fine, settings are good and permissions are good. Redis server is down.

And now I want to start Valkey via: `sudo systemctl start valkey-server`. And it starts, however the systemctl command **seems to HANG**. Checking the Valkey log says the following:

```log
systemd supervision requested or auto-detected, but Valkey is compiled without libsystemd support
```

Why is the official Ubuntu binary not compiled with libsystemd support!???? Help? I don't want to compile from source, since I think this is not how you want to deliver your product to server admins.

Kind regards,
Melroy van den Berg

---

Ps. why are the .c, .d and .o files part of the tar.gz binary?

**Comment by melroy89** [comment]
Could you make `USE_SYSTEMD=yes` part of the release binary deployment?

**Reply by roshkhatri** [reply]
I am not sure if we should be building the release binaries with `USE_SYSTEMD=yes`. 

We might want more views on this.

The README file mentions the same about building with `systemd` support.
https://github.com/valkey-io/valkey/blob/9827eef4d0c8635993224d7068afdf730d843eab/README.md?plain=1#L46

@valkey-io/core-team can you take a look.

**Reply by melroy89** [reply]
The Binary Artifacts on the download page are literally named after the Ubuntu names:

Ubuntu is using systemd, so I want to use the binary artifacts under.. well.. Ubuntu with systemd, but this setup failed! I expect the pre-build binary to be able to run on the target machine, so it can be used for production with systemd.

[arm64 / bionic](https://download.valkey.io/releases/valkey-8.0.0-bionic-arm64.tar.gz) ([sha256](https://download.valkey.io/releases/valkey-8.0.0-bionic-arm64.tar.gz.sha256))
[x86_64 / bionic](https://download.valkey.io/releases/valkey-8.0.0-bionic-x86_64.tar.gz) ([sha256](https://download.valkey.io/releases/valkey-8.0.0-bionic-x86_64.tar.gz.sha256))
[arm64 / focal](https://download.valkey.io/releases/valkey-8.0.0-focal-arm64.tar.gz) ([sha256](https://download.valkey.io/releases/valkey-8.0.0-focal-arm64.tar.gz.sha256))
[x86_64 / focal](https://download.valkey.io/releases/valkey-8.0.0-focal-x86_64.tar.gz) ([sha256](https://download.valkey.io/releases/valkey-8.0.0-focal-x86_64.tar.gz.sha256))

**Reply by zuiderkwast** [reply, marked as answer]
Sounds reasonable. It is statically linked. Do we build this binary with TSL?

I believe TLS and systemd are basic. Those who want a binary without these can build from source.

@roshkhatri What's your concern?

**Reply by melroy89** [reply]
It also sounds reasonable to me :). If it's statically linked, there shouldn't be any platform specific issues.

Ps. Is there a reason why you have bionic and focal binaries, but no Noble yet? Does the binaries even matter if they are statically build? Or are they just use newer GCC & Glibc versions?

**Reply by melroy89** [reply]
OK.. I created a PR: https://github.com/valkey-io/valkey/pull/1107, this will introduce systemd support.

**Reply by melroy89** [reply]
SO a bit off-topic but maybe a good question anyways: Should we include Noble as well or? 

My question: "Is there a reason why you have bionic and focal binaries, but no Noble yet? Does the binaries even matter if they are statically build? Or are they just use newer GCC & Glibc versions?" still remains.

**Reply by roshkhatri** [reply]
I think there is a PR open for it. I had reviewed it. Let me find out of its merged

**Reply by roshkhatri** [reply]
https://github.com/valkey-io/valkey/pull/971

**Reply by zuiderkwast** [reply]
> It is statically linked.

I guess I was wrong. When building with `make USE_SYSTEMD=yes BUILD_TLS=yes` and check `objdump -p src/valkey-server`, I get

```
Dynamic Section:
  NEEDED               libm.so.6
  NEEDED               libsystemd.so.0
  NEEDED               libssl.so.3
  NEEDED               libcrypto.so.3
  NEEDED               libc.so.6
```

This means it's dynamically loaded. It can't start if systemd and libssl, etc. are not installed.

This means systems without systemd can't use our binary releases? How many systems have OpenSSL installed but not systemd?

**Comment by roshkhatri** [comment]
>Ps. why are the .c, .d and .o files part of the tar.gz binary?

Yeah we should be fixing that.

**Reply by melroy89** [reply]
Problem line: https://github.com/valkey-io/valkey/blob/unstable/.github/workflows/call-build-linux-x86-packages.yml#L59C10-L59C50

**Reply by melroy89** [reply]
I did it myself. PR created: https://github.com/valkey-io/valkey/pull/1106

**Comment by roshkhatri** [comment]
Thank you so much @melroy89

**Reply by zuiderkwast** [reply]
I merged both PRs, since you @roshkhatri approved them. These workflows are only triggered manually anyway.

**Reply by roshkhatri** [reply]
Yes, I was kindof still reviewing the https://github.com/valkey-io/valkey/pull/1107/  PR, to see if it works. Which it should.

>These workflows are only triggered manually anyway.

These workflows are triggered automatically when a new release is published.

**Reply by roshkhatri** [reply]
Just FYI, I have tested both the changes in workflow on my fork to trigger automatically on a release 
https://github.com/roshkhatri/valkey/actions/runs/11149196519

**Reply by zuiderkwast** [reply]
> I was kindof still reviewing

but you had already approved it :D

**Reply by melroy89** [reply]
You're welcome! Small follow up question. Will the tar.gz on https://valkey.io/download/ be updated as well? Or should I wait on a patch release?

**Reply by roshkhatri** [reply]
It will not be updated. You would have to wait for a patch, which should be soon

**Reply by melroy89** [reply]
OK clear. I will wait.

> but you had already approved it :D

No worries, luckily I have enough experience to avoid regression. 😆

**Reply by roshkhatri** [reply]
>but you had already approved it :D

@zuiderkwast I had not approve it yet 😆, but its okay. No worries. i didn't see any issue anyways. the PR was pretty good.

![Screenshot 2024-10-02 at 12 42 24 PM](https://github.com/user-attachments/assets/2aab288e-d874-42a2-a01d-9be932eaa351)

**Reply by zuiderkwast** [reply]
Oh, sorry! :D haha, but you approved the other one (?)

**Reply by roshkhatri** [reply]
Yess

**Reply by melroy89** [reply]
Should I propose to setup a Matrix room for chat? 😆  I was missing that anyways for Valkey.

**Reply by roshkhatri** [reply]
Hahaha. we do have slack which is great and discord which is not very active

**Reply by melroy89** [reply]
Would love to join, but I couldn't find Slack here: https://valkey.io/connect/ (Discord is even more proprietary)

**Reply by roshkhatri** [reply]
Thats a good Idea.

**Reply by roshkhatri** [reply]
Also the new patches are now released for 8.0.1 and 7.2.7 with this changes and other CVEs. Try them and let me know if that helped.

**Reply by melroy89** [reply]
I will tomorrow thanks! It's 1:30 am now.

**Reply by melroy89** [reply]
v8.0.1 looks great! Systemd support out of the box. And no more .c, .o, etc. files.. I can continue to finish by blog post now, without explaining how to build from source.

**Reply by roshkhatri** [reply]
Thats great! Looking forward to read the blog post
